src/java.base/share/native/launcher/main.c

```diff
@@ -202,6 +231,13 @@ main(int argc, char **argv)
         JLI_List_add(args, NULL);
         margv = args->elements;
     }
+
+    if (is_checkpoint && 1 == getpid()) {
+        if (0 < fork()) {
```

Comment:
Not an expert in this area, but don't we also need to handle `SIGTERM` appropriately by forwarding it to the child process if the forking receives it?

Reply:
@DanHeidinga 
Do you mean "we need to terminate children if the parent process is terminated"?
It seems like termination of a process with PID=1 running inside a container will stop a container run, so I guess children will be terminated also. I'd appreciate any reproducible scenario, in case I'm mistaken.

Reply:
Indeed. Since this process was supposed to be the java process, it should be a proxy a good proxy for the real java process. E.g. it should forward SIGTERM and other signals (SIGQUIT to print stack trace, etc). Although this proxy should not necessarily be a good init process.

We had the same problem in restore, this code probably will be useful https://github.com/openjdk/crac/blob/crac/src/java.base/unix/native/criuengine/criuengine.c#L263